### PR TITLE
JM-6815: Update 0006 - holiday.sql

### PR DESCRIPTION
**Pull request by CCHAPMAN01**
Jira link (if applicable)

https://centralgovernmentcgi.atlassian.net/browse/JM-6815

### Change description ###
- remove "owner" column name from insert, replaced with loc_code
- update logic to create a holiday for each satellite court if a non working day is migrated for a primary court specifically
- update query to calculate expected count

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines

Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: Postgres old schema to new schema migration/0006 - holiday.sql [hunk 1]
```diff
@@ -49,7 +49,11 @@ insert into temp_bankholidays (bank_holiday,description) values ('2025/12/26', '
 update	juror_mod.migration_log
 set		source_count = (select count(1) from juror.holidays),
 		expected_target_count = (select count(1) from temp_bankholidays) +
```
Comment (ChrisWCGI) [inline Postgres old schema to new schema migration/0006 - holiday.sql:51]: What rows will be in temp_bank_holidays?

File: Postgres old schema to new schema migration/0006 - holiday.sql [hunk 2]
```diff
@@ -58,20 +62,22 @@ truncate juror_mod.holiday restart identity cascade;
...
-			tbh.bank_holiday, 
-			tbh.description,
-			true as public
-	from temp_bankholidays tbh
+	select  	null as loc_code, 
+				tbh.bank_holiday, 
+				tbh.description,
+				true as public
+	from 		temp_bankholidays tbh
 	union
```
Comment (ChrisWCGI) [inline Postgres old schema to new schema migration/0006 - holiday.sql:70]: if holidays contains dates that are also in temp_bankholidays won't we get those dates listed twice - once as public and once as not public?

File: Postgres old schema to new schema migration/0006 - holiday.sql (hunk 1 above)
Comment (CCHAPMAN01) [reply Postgres old schema to new schema migration/0006 - holiday.sql:51]: That's a great point at the moment it only contains 2024 and 2025 public holidays - how many years do we need to backdate? last 7 years to 2017?

Comment (ChrisWCGI) [reply Postgres old schema to new schema migration/0006 - holiday.sql:51]: Would likely get away with less than 7 years but I suggest we play safe and go back to 2017.

Comment (CCHAPMAN01) [reply Postgres old schema to new schema migration/0006 - holiday.sql:51]: sure will add them in 👍

**Review by ChrisWCGI — APPROVED** [review]

**Pull request merged**